ibis/backends/duckdb/tests/test_client.py

```diff
@@ -403,3 +404,23 @@ def test_read_csv_with_types(tmp_path, input, all_varchar):
     path.write_bytes(data)
     t = con.read_csv(path, all_varchar=all_varchar, **input)
     assert t.schema()["geom"].is_geospatial()
+
+
+def test_tables_accessor_no_reference_cycle():
+    """Test that a single reference to a connection has the desired lifetime semantics."""
+    con = ibis.duckdb.connect()
+
+    before = len(gc.get_referrers(con))
+    tables = con.tables
+    after = len(gc.get_referrers(con))
+
+    assert after == before
+
+    # valid call, and there are no tables in the database
+    assert not list(tables)
+
+    del con
+
+    # no longer valid because the backend has been manually decref'd
+    with pytest.raises(ReferenceError):
+        list(tables)
```

Comment:
I think the answer is no, but can we set the message that gets raises by the `ReferenceError`? Something like "The connection associated with this table set has been closed."

Reply:
The error is somewhat reasonable IMO. I'd rather avoid raising another exception if possible.

Here's the error:

```
>>> import ibis
>>> con = ibis.duckdb.connect()
>>> tables = con.tables
>>> tables
Tables
------
>>> del con
>>> tables
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cloud/src/github.com/ibis/ibis/backends/__init__.py", line 88, in __repr__
    tables = self._backend.list_tables()
ReferenceError: weakly-referenced object no longer exists
```

Reply:
I agree that it's probably a bit too low-level to be useful, but introducing a new exception to handle this seems a bit overkill. I'm open to it though if you feel strongly about it.

Reply:
One thing that's not great about introducing a new exception is that I'd have to introduce a new property to catch all `ReferenceError`s and re-raise our custom exception, and I'd have to do that everywhere a `weakref.proxy` is used.

Reply:
I don't feel strongly enough about it to merit wrapping the exception.  My hope was that there was a way of tweaking the text in the `ReferenceError` but I don't believe that's possible without catching and re-raising

Reply:
Also, I think this path is likely to be relatively uncommon

Reply:
Removed weakrefs altogether.